app/core/reservationbook_test.py

```diff
@@ -0,0 +1,98 @@
+from app.core.reservationbook import ReservationBook
+from app.core.room import Room
+from app.core.timeslot import Timeslot
+from app.core.user import User
+from app.mapper import ReservationMapper
+from app.mapper import TimeslotMapper
+from datetime import datetime
+
+
+def test_reservation_repeat(monkeypatch):
+    # Initialization
+    reservationList = []
+    waitingList = []
+    reservationBook = ReservationBook(reservationList, waitingList)
+
+    # Test Data
+    start_time = 9
+    end_time = 10
+    room_id = 2
+    block = 0
+    date = '2016-04-20'
+    user_id = 1
+    processed_description = ""
+    user = User(user_id, "", "")
+    room = Room(room_id, False)
+    time_slot = Timeslot(start_time, end_time, date, block, user_id)
+    repeat_amount = 2
+
+    # Expected result
+    expected_amount_of_reservation = repeat_amount + 1  # + 1 because 0 repeat_amount will still make 1 reservation
+
+    # Mock
+    def TimeslotMapper_makeNew(a, b, c, d, e):
+        return time_slot
+
+    def ignore1(_):
+        return
+
+    def ignore2(_, __):
+        return
+
+    def timeslot_id(_):
+        return 1
+
+    monkeypatch.setattr(Timeslot, 'setId', ignore2)
+    monkeypatch.setattr(TimeslotMapper, 'makeNew', TimeslotMapper_makeNew)
+    monkeypatch.setattr(TimeslotMapper, 'save', ignore1)
+    monkeypatch.setattr(TimeslotMapper, 'findId', timeslot_id)
+    monkeypatch.setattr(ReservationMapper, 'save', ignore1)
+
+    # Execute
+    reservationBook.makeRepeatedReservation(room, user, time_slot, processed_description, repeat_amount)
+
+    # Verify
+    assert len(reservationBook.getReservationList()) == expected_amount_of_reservation
+
+def test_find_total_reserved_time_for_user_for_a_given_week(monkeypatch):
+    expected_total_hours = 9
+
+    # Initialization
+    reservationList = []
+    waitingList = []
+    reservationBook = ReservationBook(reservationList, waitingList)
+
+    rows = 3
+    columns = 5
+
+    #Mock
+    def get_Timeslots(_):
+        timeslot_list = [[0 for x in range(columns)] for y in range(rows)]
```

Comment:
Change to:
timeslot_list = [[0]*columns]*rows

The other expression is just uglier, longer and declares two unused variables x,y. 😛